src/plugins/intel_gpu/src/graph/graph_optimizer/mark_runtime_skippable_nodes.cpp

```diff
@@ -20,6 +20,7 @@ void mark_runtime_skippable_nodes::run(program& p) {
         // Set gathers that might be skipped at runtime as can_be_optimized.
         // If not set, memory dependency will not work for the nodes that are skipped at runtime
         program_helpers::do_for_types<gather>(*node, [](gather_node& node){
+            node.is_runtime_skippable(true);
```

Comment:
We should set this only when the node is set as can_be_optimized finally. 
So please move this to line 40. Likewise for other nodes.

Reply:
Done.